src/dmgbuild/licensing.py

```diff
@@ -382,87 +158,119 @@
+            'Name': 'English'
+        }
+    ]
+}
 
 
-def maybe_encode(s, encoding):
+def maybe_encode(s, encoding="mac_roman"):
     if isinstance(s, bytes):
         return s
     return s.encode(encoding)
 
 
-def add_license(filename, license_info):
-    """Add a license agreement to the specified disk image file, which should
-    have been unflattened first."""
+def build_license(license_info):
+    """Add a license agreement to the specified disk image file,
+    see https://developer.apple.com/forums/thread/668084
+    """
+    # copy of the dict
+    xml = dict(ufifrezXMLtemplate)
 
-    fork = ResourceFork.from_file(filename)
-
-    default_lang = license_info.get('default-language', 'en_US')
-    default_lang_id = region_codes.get(default_lang, 0)
-
-    lpic = []
-    ndx = 1
     for language, license_data in license_info['licenses'].items():
-        if language not in region_codes:
+        if language not in language_names_map:
             raise Exception("Unknown language '" + language + "'. Valid languages are: " +
-                            ", ".join(sorted(region_codes.keys())))
-        encoding_name = get_encoder_name(language)
-        lang_id = region_codes[language]
+                            ", ".join(sorted(language_names_map.keys())))
 
-        # Japanese, Korean, SimpChinese, TradChinese
-        is_two_byte = lang_id in (14, 51, 52, 53)
+        language_name = language_names_map[language]
 
         if os.path.isfile(license_data):
             mode = 'rb' if license_data.endswith('.rtf') else 'r'
             with open(license_data, mode=mode) as f:
                 license_data = f.read()
 
         if type(license_data) == bytes and license_data.startswith(b'{\\rtf1'):
-            fork.add(Resource(b'RTF ', 5000 + ndx, (language + ' SLA').encode(),
-                              license_data))
+            licenseDataFormat = "RTF "
+
         else:
-            fork.add(TextResource(5000 + ndx, (language + ' SLA').encode(),
-                                  maybe_encode(license_data, encoding_name)))
-            fork.add(StyleResource(5000 + ndx, (language + ' SLA').encode(),
-                                   [Style(0, 12, 9, Style.Helvetica,
-                                          0, 0, (0, 0, 0))]))
+            licenseDataFormat = "TEXT"
+            license_data = maybe_encode(license_data)
+
+        if licenseDataFormat not in xml:
+            xml[licenseDataFormat] = []
+
+        xml[licenseDataFormat].append(
+            dict(
+                Attributes='0x0000',
+                Data=license_data,
+                ID='5000',
+                Name=language_name
+            )
+        )
 
         buttons = license_info.get('buttons', {}).get(language, None)
         if buttons is None:
-            buttons = default_buttons.get(lang_id, None)
+            buttons = default_buttons.get(language_name, None)
             if buttons is None:
                 buttons = default_buttons[0]
```

Comment:
Are you sure this path is correct now? Previously, looking up by 0 would have been valid; now, it should be look-up by language name (and could probably be folded into the `.get()` as a default value.

Reply:
yep, its also a dict now and not a list anymore... so no indexing... thanks